lua/nvim-tree/live-filter.lua

```diff
@@ -76,7 +76,8 @@ function M.apply_filter(node_)
     end
 
     local has_nodes = nodes and (M.always_show_folders or #nodes > filtered_nodes)
-    node.hidden = not (has_nodes or matches(node))
+    local ok, is_match = pcall(matches, node)
```

Comment:
We could optimise a bit to prevent this pcall when `has_nodes` however the readability is preferable.